### PR TITLE
[WEB-63] Update/web 63 new tutorial page

### PR DESCRIPTION
**Pull request by fliptopbox**
This is mostly just refactoring the existing textile articles to accommodate the metadata for the updated tutorials UI re-design.

WARNING there are a lot of files, EVERY textile article was touched

- update the front matter meta data of existing articles
- created textile articles fro externally linked assets
- created a **tutorials.json** to abstract the UI output and decouple it from website repo

### CODE REVIEW

**Comment by fliptopbox** [comment]
@jamienewcomb  and @Srushtika please review this so we can merge, thanks.

**Comment by fliptopbox** [comment]
@Srushtika I am seeing a broken portrait image for /images/tutorials/yemiwebby.png

**Comment by Srushtika** [comment]
@fliptopbox re your comments:
- I looked into the repo, the image for yemiwebby seems to have been removed somehow. It's not there in the folder as you say and I'm not sure how that happened. However, I have another image we can add in now, so it should be fine.
- This needs to be the link from that card: https://medium.com/hackernoon/building-real-time-web-applications-using-nest-js-and-ably-d85887e81f06

File: content/tutorials/_live-flight-tracking.textile [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+title: Live flight tracking app
+excerpt: Build a live flight tracker in React Native using an open data stream from Ably Hub
+section: tutorials
+category:
```
Comment (Srushtika) [inline content/tutorials/_live-flight-tracking.textile]: not sure how this got missed in the sheet but this needs to have a 'channels' tag as well

File: content/tutorials/trainapp-fitbit.textile [hunk 2]
```diff
@@ -3,7 +3,10 @@ title: Building a Clockface app for Fitbit using Ably
 alt_title: "Reactor Events Tutorial: Fitbit App using TFL arrivals data"
 excerpt: Learn how to build a FitBit App using the Ably TFL data stream and Google location data.
 section: tutorials
-category: external-services
+category:
```
[the remark this answers is not in the document]
Comment (Srushtika) [reply content/tutorials/trainapp-fitbit.textile]: Guess it's just a versioning problem, this doesn' appear in the api-hub tab either for now

File: content/tutorials/tutorials.json [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+  "categories": [
+    {
+      "id": "channels",
+      "icon": "arrows-horizontal",
+      "name": "Channels",
+      "desc": "A selection of tutorials demonstrating the use of Realtime and/or REST libraries within devices such as browsers, desktops, mobiles or servers."
+    },
+
+  "categories": [
```
Comment (Srushtika) [inline content/tutorials/tutorials.json]: Looks like this bit is repeated (which might be causing the PR deployment to fail)

File: content/tutorials/android-push-notifications.textile [hunk 4]
```diff
@@ -19,9 +22,13 @@ level: easy
...
 languages:
   - android
+  - java
+  - nodejs
+  - javascript
 ably_product: push-notifications
 tags:
-  - ably-features
+  - Android
+  - Push notification
```
Comment (Srushtika) [inline content/tutorials/android-push-notifications.textile]: Can we please replace all occurrences of the tag name 'Push notification' with 'Push Notifications'

File: content/tutorials/channel-rewind.textile [hunk 5]
```diff
@@ -14,6 +16,8 @@ languages:
   - javascript
 level: easy
 reading_time: 5
+tags:
+    - Channel rewind
```
Comment (Srushtika) [inline content/tutorials/channel-rewind.textile]: Can we please replace all occurrences of the tag name 'Channel rewind' with 'Channel Rewind'

File: content/tutorials/history.textile [hunk 6]
```diff
@@ -26,10 +28,14 @@ languages:
   - php
   - ruby
   - swift
+
 level: easy
 reading_time: 10
+tags:
+    - History
```
Comment (Srushtika) [inline content/tutorials/history.textile]: This is my bad as I didn't mention so in the sheet but can we please replace all occurrences of the tag 'History' with 'Message History'?

File: content/tutorials/mqtt-snake.textile [hunk 7]
```diff
@@ -18,7 +20,9 @@ languages:
   - python
 ably_product: adapters
 tags:
-  - ably-features
+    - MQTT
+    - Multiplayer game
+    - Protocol adapter
```
Comment (Srushtika) [inline content/tutorials/mqtt-snake.textile]: Can we please replace all occurrences of the tag name 'Protocol adapter' with 'Protocol Adapter'?

File: content/tutorials/queue-amqp-neutrino-profanity.textile [hunk 8]
```diff
@@ -16,7 +19,8 @@ languages:
   - nodejs
 ably_product: reactor
 tags:
-  -ably-features
+    - Reactor integration
```
Comment (Srushtika) [inline content/tutorials/queue-amqp-neutrino-profanity.textile]: Can we please replace all occurrences of the tag name 'Reactor integration' with 'Reactor Integration'

File: content/tutorials/live-bitcoin-pricing.textile [hunk 9]
```diff
@@ -3,7 +3,10 @@ title: Building a live bitcoin pricing chart in Angular using Ably and Kendo UI
 alt_title: "Ably <> Kendo UI"
 excerpt: Building a live bitcoin pricing chart in Angular using Ably and KendoUI
 section: tutorials
-category: libraries-integration
+category:
+    - channels
+    - api-hub
```
[the remark this answers is not in the document]
Comment (fliptopbox) [reply content/tutorials/live-bitcoin-pricing.textile]: Hi I think there is misunderstanding here .... "api-hub" is not the name of a *tag* it is the metadata category for an article. It's the JSON id for the tab, whose name is "API Streamer and Hub". So it doesn't anywhere appear the UI.

[the remark this answers is not in the document]
Comment (fliptopbox) [reply content/tutorials/live-bitcoin-pricing.textile]: > 'api-streamer' tag to all the articles currently tagged as 'api-hub'
These articles are already in the API Streamer and HUB tab, so I think tagging them is confusing for a user. Applying that filter will have no effect.

File: content/tutorials/trainapp-fitbit.textile (hunk 2 above)
[the remark this answers is not in the document]
Comment (fliptopbox) [reply content/tutorials/trainapp-fitbit.textile]: yeah, I will bump the Gemfile so it gets the latest ably/docs

File: content/tutorials/live-bitcoin-pricing.textile (hunk 9 above)
[the remark this answers is not in the document]
Comment (fliptopbox) [reply content/tutorials/live-bitcoin-pricing.textile]: The tags on articles with the same name as the tab - in which they reside - was raised in last-weeks show-and-tell and IMO it is redundant and we should avoid it. @jamienewcomb @mattheworiordan please could we align on this? @Srushtika was not on our call.

File: content/tutorials/reactor-event-zapier.textile [hunk 10]
```diff
@@ -3,7 +3,10 @@ title: "Zapier: Build an IoT realtime attendance tracker system for Slack"
 alt_title: "Reactor Events Tutorial: IoT based realtime attendance system for Slack using Zapier"
 excerpt: Learn how to build an IoT based realtime attendance system for Slack using Ably and Zapier.
 section: tutorials
-category: external-services
+category:
+    - channels
```
[the remark this answers is not in the document]
Comment (fliptopbox) [reply content/tutorials/reactor-event-zapier.textile]: please add missing metadata in sheet

File: content/tutorials/_live-flight-tracking.textile (hunk 1 above)
Comment (fliptopbox) [reply content/tutorials/_live-flight-tracking.textile]: done

File: content/tutorials/tutorials.json (hunk 3 above)
Comment (fliptopbox) [reply content/tutorials/tutorials.json]: done

File: content/tutorials/live-bitcoin-pricing.textile (hunk 9 above)
[the remark this answers is not in the document]
Comment (Srushtika) [reply content/tutorials/live-bitcoin-pricing.textile]: Sure @fliptopbox that makes sense. I was only thinking of the scenario where an article might be tagged `api-streamer` but not `api-hub`. These are two different things just so you know. Hub will always have api streamer but not the other way around

File: content/tutorials/history.textile (hunk 6 above)
Comment (fliptopbox) [reply content/tutorials/history.textile]: Done (i only found one)

File: content/tutorials/queue-amqp-neutrino-profanity.textile (hunk 8 above)
Comment (fliptopbox) [reply content/tutorials/queue-amqp-neutrino-profanity.textile]: Done (x7)

File: content/tutorials/mqtt-snake.textile (hunk 7 above)
Comment (fliptopbox) [reply content/tutorials/mqtt-snake.textile]: Done (x3)

File: content/tutorials/channel-rewind.textile (hunk 5 above)
Comment (fliptopbox) [reply content/tutorials/channel-rewind.textile]: Done (x1)

File: content/tutorials/android-push-notifications.textile (hunk 4 above)
Comment (fliptopbox) [reply content/tutorials/android-push-notifications.textile]: done (x2)

File: content/tutorials/live-bitcoin-pricing.textile (hunk 9 above)
[the remark this answers is not in the document]
Comment (fliptopbox) [reply content/tutorials/live-bitcoin-pricing.textile]: Right got you, so ...
- Which articles are in this scenario? 
- And what tag would you like to use? (api-hub is _not_ a tag)

File: content/tutorials/reactor-event-zapier.textile (hunk 10 above)
[the remark this answers is not in the document]
Comment (fliptopbox) [reply content/tutorials/reactor-event-zapier.textile]: done (x4)

File: content/tutorials/live-bitcoin-pricing.textile (hunk 9 above)
[the remark this answers is not in the document]
Comment (Srushtika) [reply content/tutorials/live-bitcoin-pricing.textile]: There are none atm. The tag would be 'API Streamer' and category would be the last tab i.e. api streamer/ hub.

[the remark this answers is not in the document]
Comment (fliptopbox) [reply content/tutorials/live-bitcoin-pricing.textile]: done

File: content/tutorials/trainapp-fitbit.textile (hunk 2 above)
[the remark this answers is not in the document]
Comment (fliptopbox) [reply content/tutorials/trainapp-fitbit.textile]: done, this should show-up when I push the updates to ably/website.

**Comment by fliptopbox** [comment]
`This branch had an error being deployed` seems to be a Heroku problem unrelated to the branch.

**Pull request merged**